galaxykit/dockerutils.py

```diff
@@ -13,42 +13,80 @@ class DockerClient:
 
     engine = ""
     registry = ""
+    tls_verify = True
 
-    def __init__(self, auth=None, engine="podman", registry="docker.io/library/"):
+    def __init__(
+        self, auth=None, engine="podman", registry="docker.io/library/", tls_verify=True
+    ):
         """
         auth should be `(username, password)` for logging into the docker registry.
 
         If not provided, then no login attempt is made.
         """
         self.engine = engine
         self.registry = registry
+        self.tls_verify = tls_verify
 
         if auth:  # we only need to auth if creds are supplied
-            run_args = [
-                engine,
-                "login",
-                registry,
-                "--username",
-                auth[0],
-                "--password",
-                auth[1],
-            ]
+            if engine == "podman":
+                run_args = [
+                    engine,
+                    "login",
+                    registry,
+                    "--username",
+                    auth[0],
+                    "--password",
+                    auth[1],
+                    f"--tls-verify={tls_verify}"  # this is only valid/necessary
+                    # with podman
```

Comment:
instead of copying all the arguments that are the same, can we do something like:

```python
run_args = [...common args...]
if engine == "podman":
    run_args.append(f"--tls-verify={tls_verify}")
```

Reply:
oh yeah, can't believe I missed that :man_facepalming: changing now.